### PR TITLE
Implement TR_J9JITaaSServerSharedCache

### PR DESCRIPTION
**Pull request by dmitry-ten**
- Initialize TR_J9JITaaSServerSharedCache on the server when
    `TR_EnableJITaaSRemoteAOT` environment variable is set.
    It will be used to make remote calls/cache information to/from
    shared class cache on the client.

 - Implement most methods that are used by shared cache on the server:
    pointerFromOffsetInSharedCache
    offsetInSharedCacheFromPointer
    persistIprofileInfo (dummy implementation, does nothing)
    rememberClass
    getClassChainOffsetOfIdentifyingLoaderForClazzInSharedCache

- Enable AOT-specific relocations that require shared cache access

### CODE REVIEW

**Comment by dmitry-ten** [comment]
This PR contains commit merged into master recently, so it should be merged after the next rebase.

**Comment by dmitry-ten** [comment]
Ready for review, but one thing that could definitely be improved is constructor.
I had to modify constructor of `TR_J9SharedCache`, because it assumes that SCC exists on the server, which is not the case for remote AOT.
Not sure what would be a better thing to do instead.

File: runtime/compiler/x/codegen/J9AheadOfTimeCompile.cpp [hunk 1]
```diff
@@ -199,7 +199,7 @@ uint8_t *J9::X86::AheadOfTimeCompile::initializeAOTRelocationHeader(TR::Iterated
          cursor += SIZEPOINTER;
 
          TR_OpaqueClassBlock *j9class = (TR_OpaqueClassBlock *) aconstNode->getAddress();
-         uintptrj_t classChainOffsetInSharedCache = sharedCache->lookupClassChainOffsetInSharedCacheFromClass(j9class);
+         uintptrj_t classChainOffsetInSharedCache = sharedCache->getClassChainOffsetOfIdentifyingLoaderForClazzInSharedCache(j9class);
```
Comment (mpirvu) [inline runtime/compiler/x/codegen/J9AheadOfTimeCompile.cpp]: Why the change in name from `lookupClassChainOffsetInSharedCacheFromClass` to `getClassChainOffsetOfIdentifyingLoaderForClazzInSharedCache`

File: runtime/compiler/env/J9SharedCache.cpp [hunk 2]
```diff
@@ -797,3 +771,47 @@ TR_J9SharedCache::getClassChainOffsetOfIdentifyingLoaderForClazzInSharedCache(TR
    uintptrj_t classChainOffsetInSharedCache = (uintptrj_t) offsetInSharedCacheFromPointer(classChainIdentifyingLoaderForClazz);
    return classChainOffsetInSharedCache;
    }
+
+void *
+TR_J9JITaaSServerSharedCache::pointerFromOffsetInSharedCache(void *offset)
+   {
```
Comment (mpirvu) [inline runtime/compiler/env/J9SharedCache.cpp]: Maybe we should store the stream into `TR_J9JITaaSServerSharedCache` object so that we don't have to call a TLS routine every time.

File: runtime/compiler/x/codegen/J9AheadOfTimeCompile.cpp (hunk 1 above)
Comment (dmitry-ten) [reply runtime/compiler/x/codegen/J9AheadOfTimeCompile.cpp]: `lookupClassChainOffsetInSharedCacheFromClass` was not present in mainline.
When I added it, `getClassChainOffsetOfIdentifyingLoaderForClazzInSharedCache` is the name Irwin and I agreed on.

File: runtime/compiler/env/J9SharedCache.cpp (hunk 2 above)
Comment (dmitry-ten) [reply runtime/compiler/env/J9SharedCache.cpp]: Will do

**Comment by dmitry-ten** [comment]
Done
